server/src/app.js

```diff
@@ -1,62 +1,76 @@
-import express from "express";
 import body_parser from "body-parser";
-import expressGraphQL from "express-graphql";
 import compression from "compression";
 import cors from "cors";
+import express from "express";
+import expressGraphQL from "express-graphql";
 import depthLimit from "graphql-depth-limit";
 
+import { connect_db, get_db_connection_status } from "./db_utils.js";
 import { create_models, create_schema } from "./models/index.js";
-import { connect_db } from "./db_utils.js";
 import {
   convert_GET_with_compressed_query_to_POST,
   get_log_object_for_request,
 } from "./server_utils.js";
 
 create_models();
-connect_db();
+
+connect_db().catch((err) => {
+  console.error(err);
+  // just logging, not trying to recover here. DB connection is reattempted per-request below
+});
 
 const schema = create_schema();
 const app = express();
 
 app.use(body_parser.json({ limit: "50mb" }));
 app.use(compression());
-app.use(cors());
+app.use(
+  cors({
+    origin: "*",
+    methods: ["POST", "GET"],
+    allowedHeaders: [
+      "Content-Type",
+      "Authorization",
+      "Content-Length",
+      "X-Requested-With",
+      "encoded-compressed-query",
+    ],
+  })
+);
 
-app.use("/", function (req, res, next) {
-  try {
-    res.header("cache-control", "public, max-age=31536000");
-    res.header("Access-Control-Allow-Origin", "*");
-    res.header(
-      "Access-Control-Allow-Headers",
-      "Content-Type, Authorization, Content-Length, X-Requested-With, encoded-compressed-query"
-    );
-    if (req.method === "OPTIONS") {
```

Comment:
Might be a stupid question but why was this here and why are you removing it?

Reply:
Haha, I think it was there for a stupid reason. That or it was a totally out of date reason. This was all CORS header configuration, and handling CORS pre-flight requests (`OPTIONS`). Well, a couple lines up we had `email_backend.use(cors())`... and I happened to glance at the docs for that `cors` middleware during this PR and realized that:
  A) by default, it doesn't call `next()` after handling `OPTIONS` requests, so those were never reaching this `if` block
  B) it has config parameters for all the CORS headers we were manually setting here (plus, if `OPTIONS` never get past it, we were never actually properly sending these headers in the pre-flight response)